pkg/scope/health/health.go

```diff
@@ -0,0 +1,150 @@
+		if err := client.Get(ctxWithTimeout, workloadObjectRef, &workloadObject); err != nil {
+			return errors.Wrapf(err, errNoWorkload, workloadRef.Name)
+		}
+
+		// TODO(asouza): Need to find a way to make sure that this is compatible with the scope structure.
+		if value, err := fieldpath.Pave(workloadObject.UnstructuredContent()).GetValue("status.resources"); err == nil {
+			refs := value.([]interface{})
+			for _, item := range refs {
+				ref := item.(map[string]interface{})
+				resourceRef := runtimev1alpha1.TypedReference{
+					APIVersion: fmt.Sprintf("%v", ref["apiVersion"]),
+					Kind:       fmt.Sprintf("%v", ref["kind"]),
+					Name:       fmt.Sprintf("%v", ref["name"]),
+				}
+
+				resourceRefs = append(resourceRefs, resourceRef)
+			}
+		} else {
+			return errors.Wrapf(err, errNoWorkloadResources, workloadRef.Name)
+		}
+	}
+
+	statusc := resourcesHealthStatus(ctxWithTimeout, log, client, healthScope.Namespace, resourceRefs)
+	status := true
+	for r := range statusc {
+		status = status && r
+	}
+
+	health := "unhealthy"
+	if status {
+		health = "healthy"
+	}
+
+	healthScope.Status.Health = health
+	return nil
+}
+
+func resourcesHealthStatus(ctx context.Context, log logging.Logger, client client.Client, namespace string, refs []runtimev1alpha1.TypedReference) <-chan bool {
+	status := make(chan bool, len(refs))
+	var wg sync.WaitGroup
+	wg.Add(len(refs))
+	for _, ref := range refs {
+		go func(resourceRef runtimev1alpha1.TypedReference) {
+			defer wg.Done()
+			err := resourceHealthStatus(ctx, client, namespace, resourceRef)
+			status <- (err == nil)
+			if err != nil {
+				log.Debug("Unhealthy resource", "resource", resourceRef.Name, "error", err)
+			}
+		}(ref)
+	}
+	go func() {
+		wg.Wait()
+		close(status)
+	}()
+
+	return status
+}
+
+func resourceHealthStatus(ctx context.Context, client client.Client, namespace string, ref runtimev1alpha1.TypedReference) error {
+	if ref.GroupVersionKind() == apps.SchemeGroupVersion.WithKind("Deployment") {
+		return deploymentHealthStatus(ctx, client, namespace, ref)
+	}
+
+	// TODO(artursouza): add other health checks.
+	// Generic health check by validating if the resource exists.
+	object := unstructured.Unstructured{}
+	object.SetAPIVersion(ref.APIVersion)
+	object.SetKind(ref.Kind)
+	objectRef := types.NamespacedName{Namespace: namespace, Name: ref.Name}
+	err := client.Get(ctx, objectRef, &object)
+	return err
+}
+
+func deploymentHealthStatus(ctx context.Context, client client.Client, namespace string, ref runtimev1alpha1.TypedReference) error {
+	deployment := apps.Deployment{}
+	deployment.APIVersion = ref.APIVersion
+	deployment.Kind = ref.Kind
+	deploymentRef := types.NamespacedName{Namespace: namespace, Name: ref.Name}
+	if err := client.Get(ctx, deploymentRef, &deployment); err != nil {
+		return errors.Wrapf(err, errResourceNotFound, ref.APIVersion, ref.Kind, ref.Name)
+	}
+
+	if deployment.Status.ReadyReplicas == 0 {
```

Comment:
deployment has many better indicators than this, ie. `DeploymentCondition` or `AvailableReplicas == Replicas`